lib/modules.nix

```diff
@@ -430,10 +430,16 @@ rec {
 
       # an attrset 'name' => list of unmatched definitions for 'name'
       unmatchedDefnsByName =
-        # Propagate all unmatched definitions from nested option sets
-        mapAttrs (n: v: v.unmatchedDefns) resultsByName
-        # Plus the definitions for the current prefix that don't have a matching option
-        // removeAttrs defnsByName' (attrNames matchedOptions);
+        if configs == []
```

Comment:
I think this could even be lifted up one level to the `unmatchedDefns` definition

Reply:
Done. Will save another few thunks and such.